cub/cub/device/dispatch/dispatch_unique_by_key.cuh

```diff
@@ -213,7 +213,9 @@ template <typename KeyInputIteratorT,
           typename NumSelectedIteratorT,
           typename EqualityOpT,
           typename OffsetT,
-          typename SelectedPolicy = DeviceUniqueByKeyPolicy<KeyInputIteratorT, ValueInputIteratorT>>
+          typename SelectedPolicy =
+            detail::unique_by_key::policy_hub<typename std::iterator_traits<KeyInputIteratorT>::value_type,
+                                              typename std::iterator_traits<ValueInputIteratorT>::value_type>>
```

Comment:
In CUB we have a shortcut:
```suggestion
          typename SelectedPolicy =
            detail::unique_by_key::policy_hub<detail::value_t<KeyInputIteratorT>,
                                              detail::value_t<ValueInputIteratorT>>>
```

Reply:
Also, should not use `std::iterator_traits` because that's not available with NVRTC, but we have to in some cases.  @miscco will fix that at some point. Using CUB's `value_t` avoids the headache.